### PR TITLE
Laboratory work #7, Darya Krinochkina - 20FPL1

### PR DESCRIPTION
**Pull request by DashaKrinochkina**

### CODE REVIEW

File: lab_7_llm/main.py [hunk 1]
```diff
@@ -6,6 +6,8 @@
 from pathlib import Path
 from typing import Iterable, Iterator, Sequence
 
+# for pull-request
```
Comment (SalnikovIgor) [inline lab_7_llm/main.py]: Remove

**Comment by SalnikovIgor** [comment]
Necessary implementation of loading your dataset.

**Comment by SalnikovIgor** [comment]
It's time to write the loading of your dataset.

**Comment by SalnikovIgor** [comment]
Fix tests

File: lab_7_llm/main.py [hunk 2]
```diff
@@ -192,4 +204,4 @@ def run(self) -> dict | None:
 
         Returns:
             dict | None: A dictionary containing information about the calculated metric
-        """
+        """
```
Comment (SalnikovIgor) [inline lab_7_llm/main.py:255]: New line

File: lab_7_llm/settings.json [hunk 3]
```diff
@@ -1,8 +1,8 @@
...
-    "model": "",
-    "dataset": "",
+    "model": "mrm8488/bert-small2bert-small-finetuned-cnn_daily_mail-summarization",
+    "dataset": "cnn_dailymail",
     "metrics": []
   },
-  "target_score": 0
-}
+  "target_score": 4
+}
```
Comment (SalnikovIgor) [inline lab_7_llm/settings.json:8]: New line

**Review by SalnikovIgor — CHANGES_REQUESTED** [review]
Fix comments and implement model analysis

**Review by SalnikovIgor — CHANGES_REQUESTED** [review]
Fix past comments

File: lab_7_llm/main.py [hunk 4]
```diff
@@ -148,6 +199,17 @@ def infer_sample(self, sample: tuple[str, ...]) -> str | None:
         Returns:
             str | None: A prediction
         """
+        tokenizer = AutoTokenizer.from_pretrained(self._model_name)
+
+        tokens = tokenizer(sample[0], max_length=120, padding=True,
+                           return_tensors='pt', truncation=True)
+        print(tokens.keys())
```
Comment (SalnikovIgor) [inline lab_7_llm/main.py]: Delete

**Review by SalnikovIgor — CHANGES_REQUESTED** [review]
Fix comments and implemetnt dataset inference

File: lab_7_llm/main.py [hunk 5]
```diff
@@ -148,6 +199,16 @@ def infer_sample(self, sample: tuple[str, ...]) -> str | None:
         Returns:
             str | None: A prediction
         """
+        tokenizer = AutoTokenizer.from_pretrained(self._model_name)
```
Comment (SalnikovIgor) [inline lab_7_llm/main.py]: Init `tokenizer` as class parameter

**Review by SalnikovIgor — CHANGES_REQUESTED** [review]

File: lab_7_llm/main.py (hunk 5 above)
Comment (DashaKrinochkina) [reply lab_7_llm/main.py]: done

**Pull request closed without merging**